crnk-gen/crnk-gen-openapi/src/main/java/io/crnk/gen/openapi/internal/schemas/AbstractSchemaGenerator.java

```diff
@@ -8,18 +9,28 @@
 
   protected final MetaResource metaResource;
 
+  protected final MetaAttribute metaAttribute;
+
   private final String prefix;
 
   protected AbstractSchemaGenerator() {
     this.metaResource = null;
+    this.metaAttribute = null;
     prefix = "";
   }
 
   protected AbstractSchemaGenerator(MetaResource metaResource) {
     this.metaResource = metaResource;
+    this.metaAttribute = null;
     prefix = metaResource.getResourceType();
   }
 
+  protected AbstractSchemaGenerator(MetaResource metaResource, MetaAttribute metaAttribute) {
+    this.metaResource = metaResource;
+    this.metaAttribute = metaAttribute;
+    prefix = metaResource.getResourceType() + metaAttribute.getName();
```

Comment:
General note: I'd ideally like to see consistent use of capitalization throughout for naming of components, ideally `PascalCase`. Currently there is a mix of `camelCase`, `PascalCase` and `notquiteCamelCase`

- `simpleTasksPostResourceReference` -> `SimpleTasksPostResourceReference`
- `simpleTasksidResourceAttribute` -> `SimpleTasksIdResourceAttribute`
- `simpleTasksnameResourceAttribute` -> `SimpleTasksNameResourceAttribute`
etc